core/src/main/java/org/elasticsearch/index/percolator/PercolatorFieldMapper.java

```diff
@@ -201,7 +197,8 @@ static Query toQuery(QueryShardContext context, boolean mapUnmappedFieldsAsStrin
     static QueryBuilder<?> parseQueryBuilder(QueryParseContext context, XContentParser parser) {
         context.reset(parser);
         try {
-            return context.parseInnerQueryBuilder();
+            // NORELEASE not really sure what to do here
+            return context.parseInnerQueryBuilder().get();
```

Comment:
I'm nor sure if this should error or ignore if the query hasn't been resolved to something at his point. Maybe @martijnvg has an opinion?